code/modules/mod/modules/modules_supply.dm

```diff
@@ -174,3 +188,146 @@
+	mod.wearer.cut_overlay(lightning)
+	var/angle = get_angle(mod.wearer, target)
+	mod.wearer.transform = mod.wearer.transform.Turn(angle)
+	mod.wearer.throw_at(get_ranged_target_turf_direct(mod.wearer, target, power), \
+		range = power, speed = max(round(0.2*power), 1), thrower = mod.wearer, spin = FALSE, \
+		callback = CALLBACK(src, .proc/on_throw_end, target, -angle))
+
+/obj/item/mod/module/hydraulic/proc/on_throw_end(atom/target, angle)
+	if(!mod?.wearer)
+		return
+	mod.wearer.transform = mod.wearer.transform.Turn(angle)
+
+/obj/item/mod/module/disposal_connector
+	name = "MOD disposal selector module"
+	desc = "A module that connects to the disposal pipeline, causing the user to go into their config selected disposal. \
+		Only seems to work when the suit is on."
+	icon_state = "disposal"
+	complexity = 2
+	idle_power_cost = DEFAULT_CHARGE_DRAIN * 0.3
+	incompatible_modules = list(/obj/item/mod/module/disposal_connector)
+	var/disposal_tag = NONE
+
+/obj/item/mod/module/disposal_connector/Initialize(mapload)
+	. = ..()
+	disposal_tag = pick(GLOB.TAGGERLOCATIONS)
+
+/obj/item/mod/module/disposal_connector/on_suit_activation()
+	RegisterSignal(mod.wearer, COMSIG_MOVABLE_DISPOSING, .proc/disposal_handling)
+
+/obj/item/mod/module/disposal_connector/on_suit_deactivation()
+	UnregisterSignal(mod.wearer, COMSIG_MOVABLE_DISPOSING)
+
+/obj/item/mod/module/disposal_connector/get_configuration()
+	. = ..()
+	.["disposal_tag"] = add_ui_configuration("Disposal Tag", "list", GLOB.TAGGERLOCATIONS[disposal_tag], GLOB.TAGGERLOCATIONS)
+
+/obj/item/mod/module/disposal_connector/configure_edit(key, value)
+	switch(key)
+		if("disposal_tag")
+			for(var/tag in 1 to length(GLOB.TAGGERLOCATIONS))
+				if(GLOB.TAGGERLOCATIONS[tag] == value)
+					disposal_tag = tag
+					break
+
+/obj/item/mod/module/disposal_connector/proc/disposal_handling(datum/disposal_source, obj/structure/disposalholder/disposal_holder, obj/machinery/disposal/disposal_machine, hasmob)
+	SIGNAL_HANDLER
+
+	disposal_holder.destinationTag = disposal_tag
+
+/obj/item/mod/module/magnet
+	name = "MOD loader hydraulic magnet module"
+	desc = "A powerful hydraulic electromagnet able to launch crates and lockers towards the user, and keep 'em attached."
+	icon_state = "magnet_loader"
+	module_type = MODULE_ACTIVE
+	removable = FALSE
+	use_power_cost = DEFAULT_CHARGE_DRAIN*3
+	incompatible_modules = list(/obj/item/mod/module/magnet)
+	cooldown_time = 1.5 SECONDS
+	overlay_state_active = "module_magnet"
+
+/obj/item/mod/module/magnet/on_select_use(atom/target)
+	. = ..()
+	if(!.)
+		return
+	if(istype(mod.wearer.pulling, /obj/structure/closet))
+		var/obj/structure/closet/locker = mod.wearer.pulling
+		playsound(locker, 'sound/effects/gravhit.ogg', 75, TRUE)
+		locker.forceMove(mod.wearer.loc)
+		locker.throw_at(target, range = 7, speed = 4, thrower = mod.wearer)
+		return
+	if(!(target in view(mod.wearer)) || !istype(target, /obj/structure/closet))
```

Comment:
switch these you want the cheapest check to be the first one executed so that the more expensive check doenst run if it cant change the answer